src/crimson/mon/MonClient.cc

```diff
@@ -929,6 +929,16 @@ static entity_addr_t choose_client_addr(
 seastar::future<bool> Client::reopen_session(int rank)
 {
   logger().info("{} to mon.{}", __func__, rank);
+  if (!is_hunting()) {
```

Comment:
nit, 

1. might want to check `is_hunting()` instead of `!is_hunting()` for better readability, like
  ```c++
  if (is_hunding()) {
    // close active_con
  } else {
    // close pending_con
  }
  ```
2. remove the cleanup code in `ms_handle_reset()` which close and reset `active_con` also.

Reply:
> might want to check is_hunting() instead of !is_hunting() for better readability

`is_hunting()` is just `!active_con` and it looks it's used rather sparingly.  I'll test `active_con` directly.

> remove the cleanup code in ms_handle_reset() which closes and resets active_con also.

Done.